File: Assignment5/app/src/main/java/com/example/assignment5/Ques4/MainActivity.java
package com.example.assignment5.Ques4;

import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentTransaction;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import com.example.assignment5.R;

public class MainActivity extends AppCompatActivity implements ListFragment.Communication {
    FragmentManager fragmentManager;
    ListFragment listFragment;
    OutputFragment outputFragment;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        fragmentManager=getSupportFragmentManager();
        listFragment=new ListFragment();
        FragmentTransaction fragmentTransaction=fragmentManager.beginTransaction();
        fragmentTransaction.add(R.id.listFragment, listFragment, null);
        fragmentTransaction.commit();

    }


    @Override
    public void communicate(Bundle  bundle) {
        outputFragment=new OutputFragment();
        outputFragment.getData(bundle);
        FragmentTransaction fragmentTransaction1=fragmentManager.beginTransaction();
        fragmentTransaction1.add(R.id.outputFragment, outputFragment, null);
    }
}

File: Kotlin4/KotlinLogin/app/src/main/java/com/example/kotlinlogin/MainActivity.kt
package com.example.kotlinlogin

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.text.TextUtils
import android.util.Patterns
import android.widget.Button
import android.widget.EditText
import android.widget.Toast

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        var usernameET=findViewById<EditText>(R.id.usernameET)
        var passwordET=findViewById<EditText>(R.id.passwordET)
        var button=findViewById<Button>(R.id.loginBtn)




        button.setOnClickListener {
            if(!validateEmail(usernameET.text.toString())){
                    Toast.makeText(this, "Invalid Email",Toast.LENGTH_LONG).show()
                }
            else{
                Toast.makeText(this, "Details Submitted", Toast.LENGTH_LONG).show()
                usernameET.setText("")
                passwordET.setText("")
            }

        }
    }

    fun validateEmail(str:String): Boolean {
        return (!TextUtils.isEmpty(str) && Patterns.EMAIL_ADDRESS.matcher(str).matches())
    }
}
